src/apiCalls.js

```diff
@@ -108,20 +117,35 @@ const updateCurrentUser = (user) => {
 
 const postRecipeToCook = (userID, recipeID, e) => {
   updateRecipe(userID, recipeID, "POST")
-    .then((res) => {
-      if (res.message) {console.log("post response", res.message)}
-      getUsersAfterUpdate(userID, recipeID, e);
+    .then((res) => res.json())
+    .then(status => {
+      if (status.message.includes("was added")) {
+        getUsersAfterUpdate(userID, recipeID, e);
+      } else {
+        showError(recipeID)
+      }
     })
-    .catch(err => console.error(err));
+    .catch(err => {
+      showError(recipeID);
+      console.error(err);
+    });
 }
 
 const deleteRecipeToCook = (userID, recipeID, e) => {
```

Comment:
I just realized these to functions are almost identical. Can we dry this up with the following dynamic function replacing them? 

```
const updateServerRecipe = (userID, recipeID, e, requestType) => {
  const conditions = {
    "POST": 'added',
    "DELETE": 'removed'
  }
  updateRecipe(userID, recipeID, requestType)
  .then((res) => res.json())
  .then(status => {
    if (status.message.includes(`was ${conditions[requestType]}`)) {
      getUsersAfterUpdate(userID, recipeID, e);
    } else {
      showError(recipeID)
    }
  })
  .catch(err => {
    showError(recipeID);
    console.error(err);
  });
}
```

"POST" or "DELETE" would be passed in as the request type upon invokation

Reply:
Done